api/internal/impl/acl/list.go

```diff
@@ -63,45 +68,29 @@ import (
 func ListPolicies(
 	source *workloadapi.X509Source,
 	SPIFFEIDPattern string, pathPattern string,
-) (*[]data.Policy, error) {
+) (*[]data.Policy, *sdkErrors.SDKError) {
 	if source == nil {
-		return nil, sdkErrors.ErrNilX509Source
+		return nil, sdkErrors.ErrSPIFFENilX509Source
 	}
 
 	r := reqres.PolicyListRequest{
 		SPIFFEIDPattern: SPIFFEIDPattern,
 		PathPattern:     pathPattern,
 	}
-	mr, err := json.Marshal(r)
-	if err != nil {
-		return nil, errors.Join(
-			errors.New(
-				"listPolicies: I am having problem generating the payload",
-			),
-			err,
-		)
+	mr, marshalErr := json.Marshal(r)
+	if marshalErr != nil {
+		failErr := sdkErrors.ErrDataMarshalFailure.Wrap(marshalErr)
+		failErr.Msg = "problem generating the payload"
+		return nil, failErr
 	}
 
-	client := net.CreateMTLSClientForNexus(source)
-
-	body, err := net.Post(client, url.PolicyList(), mr)
-	if err != nil {
-		if errors.Is(err, sdkErrors.ErrNotFound) {
-			return nil, nil
+	res, postErr := net.PostAndUnmarshal[reqres.PolicyListResponse](
+		source, url.PolicyList(), mr)
+	if postErr != nil {
+		if postErr.Is(sdkErrors.ErrAPINotFound) {
+			return &([]data.Policy{}), nil
```

Comment:
we used to return `nil` in these instances (both for secrets, secret metadata(s), and policies); returning an empty list is a more semantically-correct response. -- the resultset does not "not exist", it is just empty.